ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -594,11 +594,16 @@ public void onClose(
         final QueryMetadata query
     ) {
       final String applicationId = query.getQueryApplicationId();
+      Optional<String> topologyName = Optional.empty();
+      if (ksqlConfig.getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+        topologyName = Optional.of(query.getQueryId().toString());
+      }
       if (query.hasEverBeenStarted()) {
         cleanupService.addCleanupTask(
             new QueryCleanupService.QueryCleanupTask(
                 serviceContext,
                 applicationId,
```

Comment:
I'm probably just missing context here, but if shared runtimes is on, what's the application id here?

Reply:
it not specific to any query